qrand/platforms/cirq/backend.py

```diff
@@ -19,23 +19,71 @@
 ## WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 ## See the License for the specific language governing permissions and
 ## limitations under the License.
-
+from cirq_google.engine import Engine 
 from ..backend import QuantumBackend
 
 
 ###############################################################################
 ## CIRQ BACKEND (DECORATOR)
 ###############################################################################
-class CirqBackend(QuantumBackend):
-    def __init__(self) -> None:
-        self.ERROR_MSG = f"{self.__class__.__name__}"  # TODO
-        raise NotImplementedError(self.ERROR_MSG)
+
+
+
+class CirqBackend(QuantumBackend,Engine):
+    """
+    
+        Encapsulating Google Engine API with Quantum Backend 
+
+        Since API and Cirq is still in Alpha,Changes are expected
+      
+        methods from Engine class:
+
+        create_program()->Wraps a Circuit for use with the Quantum Engine. i.e return engineprogram
+
+        run()->Runs the supplied Circuit via Quantum Engine. i.e return study.result    
+    """
+    def __init__(
+                self, 
+                engine: Engine,
+                shots:int =1,
+                max_experiments:int=1
+        ) -> None:
+        self._backend: Engine = engine
+        super(QuantumBackend, self).__init__(   )
+        self._options = engine.service_args
+        self._shots:int= shots,
+        self._max_experiments = max_experiments
+        self.max_shots=shots #Google API is still private 
 
     ############################### PUBLIC API ###############################
     @property
     def max_measurements(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return  (self._max_experiments*self._shots )
 
+    @property
+    def max_experiments(self) -> int:
+        return self._max_experiments
+ 
+    @max_experiments.setter
+    def max_experiments(self,max_experiments:int)->None:
+        self._max_experiments:int=max_experiments
+
+    """
     @property
     def max_qubits(self) -> int:
-        raise NotImplementedError(self.ERROR_MSG)
+        return (
+            
+        )
+    """
+
```

Comment:
I couldn't figure out the maximum qubits part. We could define this explicitly(Thoughts!)